find_email_addresses.py

```diff
@@ -1,59 +0,0 @@
-from bs4 import BeautifulSoup
-from collections import deque
-import requests
-import re
-import sys
-
-# Starting URL should be first argument from sys
-start_url = sys.argv[1]
-if not start_url.startswith("http://"):
-	start_url = "http://" + start_url 
-
-q = deque([start_url])
-
-urls_seen = set([])
-emails = set([])
-
-while q:
-	url = q.popleft()
-	urls_seen.add(url)
-
-	# Allow redirecting only on start_url
-	r = requests.get(url, allow_redirects = (url == start_url))
-
-	soup = BeautifulSoup(r.text, 'html.parser')
-
-	# Extract all email addresses and add them into the resulting set
-	new_emails = set(re.findall(r"[a-z0-9\.\-+_]+@[a-z0-9\.\-+_]+\.[a-z]+", r.text, re.I))
-	emails.update(new_emails)
-
-	# Iterate over all links found on the page
-	for link in soup.find_all('a'):
-
-		# It's possible to see 'a' elements without 'href'
-		try:		
-			new_url = link.attrs["href"]
-		except:
-			break
```

Comment:
I think you'd want to continue here not break, right?  Is it possible to check for href without the exception being thrown? Might be cleaner and more readable to use a conventional if check on link